tests/components/influxdb/test_sensor.py

```diff
@@ -0,0 +1,537 @@
+            {"queries_flux": [{"name": "test", "query": "{{ illegal.template }}"}]},
+            _set_query_mock_v2,
+            _make_v2_resultset,
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_error_rendering_template(
+    hass, caplog, mock_client, config_ext, queries, set_query_mock, make_resultset
+):
+    """Test behavior of sensor with error rendering template."""
+    set_query_mock(mock_client, return_value=make_resultset(42))
+
+    sensors = await _setup(hass, config_ext, queries, ["sensor.test"])
+    assert sensors[0].state == STATE_UNKNOWN
+    assert (
+        len([record for record in caplog.records if record.levelname == "ERROR"]) == 1
+    )
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, queries, set_query_mock, test_exception, make_resultset",
+    [
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            OSError("fail"),
+            _make_v1_resultset,
+        ),
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            InfluxDBClientError("fail"),
+            _make_v1_resultset,
+        ),
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            InfluxDBServerError("fail"),
+            _make_v1_resultset,
+        ),
+        (
+            API_VERSION_2,
+            BASE_V2_CONFIG,
+            BASE_V2_QUERY,
+            _set_query_mock_v2,
+            OSError("fail"),
+            _make_v2_resultset,
+        ),
+        (
+            API_VERSION_2,
+            BASE_V2_CONFIG,
+            BASE_V2_QUERY,
+            _set_query_mock_v2,
+            ApiException(),
+            _make_v2_resultset,
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_connection_error_at_startup(
+    hass,
+    caplog,
+    mock_client,
+    config_ext,
+    queries,
+    set_query_mock,
+    test_exception,
+    make_resultset,
+):
+    """Test behavior of sensor when influx returns error."""
+    query_api = set_query_mock(mock_client, side_effect=test_exception)
+    expected_sensor = "sensor.test"
+
+    # Test sensor is not setup first time due to connection error
+    await _setup(hass, config_ext, queries, [])
+    assert hass.states.get(expected_sensor) is None
+    assert (
+        len([record for record in caplog.records if record.levelname == "ERROR"]) == 1
+    )
+
+    # Stop throwing exception and advance time to test setup succeeds
+    query_api.reset_mock(side_effect=True)
+    set_query_mock(mock_client, return_value=make_resultset(42))
+    new_time = dt_util.utcnow() + timedelta(seconds=PLATFORM_NOT_READY_BASE_WAIT_TIME)
+    async_fire_time_changed(hass, new_time)
+    await hass.async_block_till_done()
+    assert hass.states.get(expected_sensor) is not None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)